### DISCUSSION TITLE
Accuracy Metric does not perform squeeze on input data

### DISCUSSION THREAD

**Discussion by jungla88** [Lightning-AI/torchmetrics#1233] [Classification] [answered]
Hello,

I am not sure if this is a bug or not but according to [https://torchmetrics.readthedocs.io/en/stable/pages/classification.html#input-types](url), prediction tensor is squeezed starting from dimension one. Accuracy seems not performing such operation and in validity checks raised an error:

`preds = torch.tensor([0.4,0.51,0.51,0.6,0.7])`
`target = torch.tensor([0,1,0,0,1])`
`acc = Accuracy(num_classes=1)`
`acc(preds,target)` 

This works ok, whilst when `acc(preds.unsqueeze(dim=1),target)`:

`ValueError: The highest label in target should be smaller than the size of the C dimension of preds.`

**Comment by Borda** [comment, marked as answer]
@jungla88 — this was a real issue with the old unified `Accuracy` class but has been resolved by the task-specific API refactor. As of v1.9.0:

**Use `BinaryAccuracy` for binary classification:**
```python
from torchmetrics.classification import BinaryAccuracy

acc = BinaryAccuracy()

preds = torch.tensor([0.4, 0.51, 0.51, 0.6, 0.7])
target = torch.tensor([0, 1, 0, 0, 1])

# Works with 1D
acc(preds, target)  # tensor(0.6000)

# Also works with extra dim — internally handled
acc(preds.unsqueeze(1), target)  # same result, no error
```

**For spatial predictions** (segmentation):
```python
acc = BinaryAccuracy(multidim_average="global")   # flatten all spatial dims
acc = BinaryAccuracy(multidim_average="samplewise")  # per-sample accuracy
```

**The old `Accuracy(num_classes=1)` pattern is deprecated.** Always use task-specific classes:

| Task | Class |
|------|-------|
| Binary | `BinaryAccuracy` |
| Multiclass | `MulticlassAccuracy(num_classes=N)` |
| Multilabel | `MultilabelAccuracy(num_labels=N)` |

Docs: [Accuracy](https://lightning.ai/docs/torchmetrics/stable/classification/accuracy.html)
